n2survey/lime/survey.py

```diff
@@ -588,3 +588,105 @@ def get_choices(self, question: str) -> dict:
+
+        Returns:
+            df: Mental health condition ratings and classifications
+        """
+
+        # set up condition-specific parameters
+        if condition == "state":
+            base_score = 10 / 3
+            num_subquestions = 6
+            conversion = ["pos", "neg", "neg", "pos", "pos", "neg"]
+            label = "state_anxiety"
+            classification_boundaries = [0, 37, 44, 80]
+            classes = ["no or low anxiety", "moderate anxiety", "high anxiety"]
+        elif condition == "trait":
+            base_score = 5 / 2
+            num_subquestions = 8
+            conversion = [
+                "pos",
+                "neg",
+                "neg",
+                "pos",
+                "neg",
+                "neg",
+                "pos",
+                "neg",
+            ]
+            label = "trait_anxiety"
+            classification_boundaries = [0, 37, 44, 80]
+            classes = ["no or low anxiety", "moderate anxiety", "high anxiety"]
+        elif condition == "depression":
+            base_score = 1
+            num_subquestions = 8
+            conversion = ["freq" for i in range(8)]
+            label = "depression"
+            classification_boundaries = [0, 4, 9, 14, 19, 24]
+            classes = [
+                "no to minimal depression",
+                "mild depression",
+                "moderate depression",
+                "moderately severe depression",
+                "severe depression",
+            ]
+        else:
+            raise ValueError(
+                "Unsupported condition type. Please consult your friendly local psychiatrist."
+            )
+
+        # Set up score conversion dicts
+        pos_direction_scores = {
+            "Not at all": 4 * base_score,
+            "Somewhat": 3 * base_score,
+            "Moderately": 2 * base_score,
+            "Very much": 1 * base_score,
+        }
+        neg_direction_scores = {
+            "Not at all": 1 * base_score,
+            "Somewhat": 2 * base_score,
+            "Moderately": 3 * base_score,
+            "Very much": 4 * base_score,
+        }
+        frequency_scores = {
+            "Not at all": 0 * base_score,
+            "Several days": 1 * base_score,
+            "More than half the days": 2 * base_score,
+            "Nearly every day": 3 * base_score,
+        }
+        conversion_dicts = {
+            "pos": pos_direction_scores,
+            "neg": neg_direction_scores,
+            "freq": frequency_scores,
+        }
+
+        # Map responses from code to text then to score
+        df = pd.DataFrame()
+        subquestion_columns = [
+            f"{question}_SQ00{str(i)}" for i in range(1, num_subquestions + 1)
+        ]
+        for column, conversion in zip(subquestion_columns, conversion):
+            df[f"{column}_score"] = (
+                self.responses[column]
+                .map(self.get_choices(question))
```

Comment:
As far as I understand, this is not needed if `self.get_responses(question)` is used

Reply:
Was not aware of the `get_responses` method. Must have been added while I was away last year. 

Decided to keep part of my original code because the `get_responses` method either returns column names AND responses both in id or both in text, and I convert the responses from id to text myself while keeping the column names as id, because I want to retain the naming convention of columns in case the results were to be concatenated to the rest of the `self.responses` df.